server/src/test/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapperTests.java

```diff
@@ -757,6 +758,8 @@ public void testIndexStoredArraySourceSingleLeafElementAndNull() throws IOExcept
     }
 
     public void testIndexStoredArraySourceSingleLeafElementInObjectArray() throws IOException {
+        roundtripMaskedFields.add("path.int_value.offsets");
```

Comment:
This is temporary, right? Until a field mapper can detect that a parent field can overwrite `synthetic_source_keep`? Because currently the synthetic_source_keep index setting determines a  field keep arrays or all.

Reply:
I think we need to have another discussion on this; I am still confused on how inheriting the `synthetic_source_keep` mode solves this issue

Reply:
> I am still confused on how inheriting the synthetic_source_keep mode solves this issue

Let me look into this.